src/charm.py

```diff
@@ -94,6 +98,54 @@ def __init__(self, *args):
         self.framework.observe(
             self.fiveg_n3_provider.on.fiveg_n3_request, self._on_fiveg_n3_request
         )
+        self.framework.observe(self.on.install, self._on_install)
+        self.framework.observe(self.on.remove, self._on_remove)
+
+    def _on_install(self, event: EventBase) -> None:
+        self._create_external_upf_service()
+
+    def _create_external_upf_service(self) -> None:
+        client = Client()
+        service = Service(
+            apiVersion="v1",
+            kind="Service",
+            metadata=ObjectMeta(
+                namespace=self._namespace,
+                name=f"{self.app.name}-external",
+                labels={
+                    "app.kubernetes.io/name": self.app.name,
+                },
+            ),
+            spec=ServiceSpec(
+                selector={
+                    "app.kubernetes.io/name": self.app.name,
+                },
+                ports=[
+                    ServicePort(name="pfcp", port=8805, protocol="UDP"),
+                ],
+                type="LoadBalancer",
```

Comment:
This will create a service with a port assigned by Loadbalancer. 
service/upf-external                          LoadBalancer   10.152.183.224   10.0.0.2      8805:31523/UDP      2m
I have a question here.
Do we need to allow to define any specific port by user ? At the moment, it will be automatically assigned by Loadbalancer service. (31523)

Second issue:
I am not sure, if we need to consider this at the moment but for public cloud environments, we may need to provide Loadbalancer IP, if cloud-provider requires to specify the loadBalancerIP.

Reply:
Ad. 1. I don't think it's a problem. What matters is that we need our service to be available on port `8805`. Which internal port it maps to doesn't really matter.
Ad. 2. I'm not sure I understand the problem. Would you mind giving an example?

Reply:
OK, that explains it :) I didn't know that option ever existed :D